File: Golang/9-palindrome-number/palindrome-number.go
func isPalindrome(x int) bool {
    
    if x < 0 {
        return false
    }
    if x < 10 {
        return true
    }

    temp := 0
    newNum := 0
    y := x
    
    for y > 0 {
        temp = y%10
        newNum = newNum*10 + temp
        y=y/10
    }
    
    if newNum == x {
        return true
    }
    return false
}

File: Python/9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        rev = 0
        temp = x
        while temp > 0:
            rev = (temp % 10) + (rev * 10)
            temp = temp // 10
        if rev == x:
            return True
        else:
            return False